src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -234,13 +237,26 @@ private function appendToken(SessionInterface $sessionContainer, Response $respo
             return FigResponseCookies::set($response, $this->getExpirationCookie());
         }
 
-        if (! $sessionContainerChanged) {
-            return $response;
+        if ($sessionContainerChanged || $this->shouldTokenBeRefreshed($token)) {
+            return FigResponseCookies::set($response, $this->getTokenCookie($sessionContainer));
+        }
+
+        return $response;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    private function shouldTokenBeRefreshed(Token $token = null) : bool
+    {
+        if (null === $token) {
+            return false;
         }
 
-        return FigResponseCookies::set($response, $this->getTokenCookie($sessionContainer));
+        return time() >= ($token->getClaim('iat') + $this->refreshTime);
```

Comment:
As discussed, the only possible outcome of this is that the user will trigger crypto token verification at each page load. Unless the user is able to forge a pseudo-real signature, this process will bail out early and have no DDoS-alike effects anyway.